src/chain.h

```diff
@@ -382,6 +382,7 @@ class CDiskBlockIndex : public CBlockIndex
 
     SERIALIZE_METHODS(CDiskBlockIndex, obj)
     {
+        LOCK(::cs_main);
```

Comment:
can you explain why this is needed?

`CDiskBlockIndex` creates a copy of the data and it isn't used in validation or blockstorage code. It is only used to serialize a struct and then discarded.

Reply:
Having a look.

Reply:
> can you explain why this is needed?
> 
> `CDiskBlockIndex` creates a copy of the data and it isn't used in validation or blockstorage code. It is only used to serialize a struct and then discarded.

Good point. Opened #24199 as an approach to address this.